interviewTest/Task2.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ This is simple Login form component.
+
+ 1. There are 3x fields - a user name (text input), a password (text input) and a 'remember me' checkbox
+ 2. All 3 fields should be editable by the user
+ 3. The form can be submitted by the user with the correct data
+
+ There are issues with this implementation. Can you see them?
+*/
+import CheckBox from "@react-native-community/checkbox";
+import React, { useState } from "react";
+import { Button, StyleSheet, Text, TextInput, View } from "react-native";
+import { isTablet } from "react-native-device-info";
+
+type Props = {
+  onSubmit: () => void;
```

Comment:
> [!CAUTION]
> `onSubmit` isn't passed any of the data that the user filled into the form. At first I thought maybe it's some kind of "notify-my-parent" function (e.g. the child submits the form and then the parent closes the modal/dialog that the child was rendered in), but after reading line 52 I realized that it wasn't so.

The fix for this is simple: pass the user's name, password and remember-me choice to the parent:
```suggestion
  onSubmit: ({
    username: string,
    password: string,
    // This should default to false.
    shouldRememberUser?: boolean
  }) => void;
```